tests/functional/replication/queueProcessor.js

```diff
@@ -362,28 +378,50 @@ describe('queue processor error management with mocking', () => {
 
         // don't call start() on the queue processor, so that we don't
         // attempt to fetch entries from kafka
-
-        s3mock = new S3Mock();
-        httpServer = http.createServer(
-            (req, res) => s3mock.onRequest(req, res));
-        httpServer.listen(7777);
     });
 
     afterEach(() => {
         s3mock.resetTest();
+        httpServer.close();
     });
 
-    after(() => {
-        httpServer.close();
+    describe('object with single part', () => {
+        before(() => {
+            s3mock = new S3Mock([locations[0]]);
+            httpServer = http.createServer(
+                (req, res) => s3mock.onRequest(req, res));
+            httpServer.listen(7777);
+        });
+
+        it('should complete a replication end-to-end', done => {
+            queueProcessor.processKafkaEntry(s3mock.kafkaEntry, err => {
+                assert.ifError(err);
+                assert(s3mock.hasPutTargetData);
+                assert(s3mock.hasPutTargetMd);
+                assert(s3mock.hasPutSourceMd);
+                done();
+            });
+        });
     });
 
-    it('should complete a replication end-to-end', done => {
-        queueProcessor.processKafkaEntry(s3mock.kafkaEntry, err => {
-            assert.ifError(err);
-            assert(s3mock.hasPutTargetData);
-            assert(s3mock.hasPutTargetMd);
-            assert(s3mock.hasPutSourceMd);
-            done();
+    describe('object with multiple parts', () => {
+        before(() => {
+            // don't call start() on the queue processor, so that we don't
```

Comment:
This comment is only meaningful when creating the kafka queue, should be removed here.